lib/ddtrace/ext/http.rb

```diff
@@ -1,14 +1,61 @@
 module Datadog
   module Ext
     module HTTP
-      TYPE = 'http'.freeze
-      TEMPLATE = 'template'.freeze
-      URL = 'http.url'.freeze
       BASE_URL = 'http.base_url'.freeze
+      ERROR_RANGE = 500...600
       METHOD = 'http.method'.freeze
-      REQUEST_ID = 'http.request_id'.freeze
       STATUS_CODE = 'http.status_code'.freeze
-      ERROR_RANGE = 500...600
+      TEMPLATE = 'template'.freeze
+      TYPE = 'http'.freeze
+      URL = 'http.url'.freeze
+
+      # Constants for request headers
+      module RequestHeaders
+        CACHE_CONTROL = 'http.request.headers.cache_control'.freeze
```

Comment:
As we discussed, probably it's better to make it configurable as an option (easy accessible from `rack` and `rails` and `sinatra`) so that users can define the list of headers they want to collect (even custom ones).

Then we have to pick the right defaults, so I'd advocate to collect what we're collecting right now (e.g. `request_id`) and then double check with our security team for the rest.

The Agent, still has the capability to redact / obfuscate automatically collected headers and is configurable as defined in https://github.com/DataDog/datadog-trace-agent/pull/398.

Sounds good @delner ?